src/k8s/pkg/k8sd/features/cilium/cleanup.go

```diff
@@ -0,0 +1,26 @@
+package cilium
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+
+	"github.com/canonical/k8s/pkg/snap"
+)
+
+func CleanupNetwork(ctx context.Context, snap snap.Snap) error {
+	if err := os.Remove("/var/run/cilium/cilium.pid"); err != nil {
+		return fmt.Errorf("failed to remove cilium pid file: %w", err)
+	}
+
+	if _, err := os.Stat("/opt/cni/bin/cilium-dbg"); err == nil {
+		if err := exec.CommandContext(ctx, "/opt/cni/bin/cilium-dbg", "cleanup", "--all-state", "--force").Run(); err != nil {
+			return fmt.Errorf("cilium-dbg cleanup failed: %w", err)
+		}
+	} else {
+		return fmt.Errorf("cilium-dbg can't be accessed: %w", err)
```

Comment:
cleanup is done on a best-effort basis. we have the `k8s .. cleanup || true` at the top-level, true. but this here could be problematic.

e.g: if the pid file does not exist, that's not a problem, it doesn't exist so we can just proceed. etc etc.